### PR TITLE
Japanese support for `get_word_splitter`

### PR DESCRIPTION
**Pull request by sarahyurick**
Closes https://github.com/NVIDIA/NeMo-Curator/issues/257.

### CODE REVIEW

File: nemo_curator/utils/text_utils.py [hunk 1]
```diff
@@ -23,16 +23,33 @@
...
     if language == "zh":
         import jieba
 
-        return jieba.cut
+        def jieba_splitter(text):
+            return list(jieba.cut(text))
+
+        return jieba_splitter
+
+    elif language == "ja":
```
Comment (ryantwolf) [inline nemo_curator/utils/text_utils.py:35]: Do you mind updating the docstrings in the filters where the `get_word_splitter` is used to explain the splitting strategy we do for each language?

File: requirements/requirements.txt [hunk 2]
```diff
@@ -12,6 +12,7 @@ in-place==0.5.0
 jieba==0.42.1
 justext==3.0.1
 lxml_html_clean
+mecab-python3==1.0.9
```
Comment (ryantwolf) [inline requirements/requirements.txt:15]: Any reason for the hard pin? If not, I'd prefer to just keep it at `mecab-python3`

**Review by ryantwolf — CHANGES_REQUESTED** [review]
With documentation updates, this should be good to go.

**Comment by sarahyurick** [comment]
Thanks @ryantwolf ! Updated.

**Review by ryantwolf — APPROVED** [review]
Great, thanks!

**Pull request merged**